pkg/reconciler/sinkbinding/sinkbinding.go

```diff
@@ -0,0 +1,194 @@
+		sb.Status.Auth = &duckv1.AuthStatus{
+			ServiceAccountName: &saName,
+		}
+
+		if err := auth.EnsureOIDCServiceAccountExistsForResource(ctx, s.serviceAccountLister, s.kubeclient, v1.SchemeGroupVersion.WithKind("SinkBinding"), sb.ObjectMeta); err != nil {
+			sb.Status.MarkOIDCIdentityCreatedFailed("Unable to resolve service account for OIDC authentication", "%v", err)
+			return err
+		}
+		sb.Status.MarkOIDCIdentityCreatedSucceeded()
+
+		err := s.reconcileOIDCTokenSecret(ctx, sb)
+		if err != nil {
+			sb.Status.MarkOIDCTokenSecretCreatedFailed("Unable to reconcile OIDC token secret", "%v", err)
+			return err
+		}
+		sb.Status.MarkOIDCTokenSecretCreatedSuccceeded()
+
+	} else {
+		sb.Status.Auth = nil
+		sb.Status.MarkOIDCIdentityCreatedSucceededWithReason(fmt.Sprintf("%s feature disabled", feature.OIDCAuthentication), "")
+		sb.Status.MarkOIDCTokenSecretCreatedSuccceededWithReason(fmt.Sprintf("%s feature disabled", feature.OIDCAuthentication), "")
+		sb.Status.OIDCTokenSecretName = nil
+	}
+
+	return nil
+}
+
+// I'm just here so I won't get fined
+func (*SinkBindingSubResourcesReconciler) ReconcileDeletion(ctx context.Context, b psbinding.Bindable) error {
+	return nil
+}
+
+func (s *SinkBindingSubResourcesReconciler) reconcileOIDCTokenSecret(ctx context.Context, sb *v1.SinkBinding) error {
+	logger := logging.FromContext(ctx)
+	secretName := fmt.Sprintf("oidc-token-%s", sb.Name)
+
+	if sb.Status.SinkAudience == nil {
+		return fmt.Errorf("sinkAudience must be set on %s/%s to generate a OIDC token secret", sb.Name, sb.Namespace)
+	}
+
+	var applyConfig *applyconfigurationcorev1.SecretApplyConfiguration
+
+	secret, err := s.secretLister.Secrets(sb.Namespace).Get(secretName)
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			// create new secret
+			apiVersion := fmt.Sprintf("%s/%s", v1.SchemeGroupVersion.Group, v1.SchemeGroupVersion.Version)
+			applyConfig = new(applyconfigurationcorev1.SecretApplyConfiguration).
+				WithName(secretName).
+				WithNamespace(sb.Namespace).
+				WithType(corev1.SecretTypeOpaque).
+				WithGeneration(1).
+				WithKind("Secret").
+				WithAPIVersion("v1").
+				WithOwnerReferences(&applyconfigurationmetav1.OwnerReferenceApplyConfiguration{
+					APIVersion:         &apiVersion,
+					Kind:               pointer.String("SinkBinding"),
+					Name:               &sb.Name,
+					UID:                &sb.UID,
+					Controller:         pointer.Bool(true),
+					BlockOwnerDeletion: pointer.Bool(false),
+				})
+
+			logger.Debugf("No OIDC token secret found for %s/%s sinkbinding. Will create a new secret", sb.Name, sb.Namespace)
+		} else {
+			return fmt.Errorf("could not check if secret %q exists already: %w", secretName, err)
+		}
+	} else {
+		// check if token needs to be renewed
+		expiry, err := auth.GetJWTExpiry(string(secret.Data["token"]))
+		if err != nil {
+			return fmt.Errorf("could not get expiry date: %w", err)
+		}
```

Comment:
Did you consider renewing the token when there is an error in parsing/validating it? 

I would expect that when someone change the topic and it's not valid for whatever reason (corrupted or whatever) the operator would self-heal itself without user intervention by renewing it, would you expect the same?

Reply:
Makes sense. Updated it :+1: